# HYPOTHESIS TESTING

**Null Hypothesis (H₀):** The theoretical mean weekly cost according to the cost model is accurate.


H
0
: μ = theoretical mean weekly cost

**Alternative Hypothesis (H₁):** The observed mean weekly cost is higher than the theoretical mean weekly cost.

H
1
​
 : μ > theoretical mean weekly cost

In [ ]:
import scipy.stats as stats
import numpy as np

In [ ]:
sample_mean = 3050  # sample mean weekly cost
theoretical_mean = 4000  # theoretical mean weekly cost
std_deviation_per_unit = 25  # standard deviation of units produced
units_coefficient = 5  # coefficient in the cost model
sample_size = 25  # number of restaurants

In [ ]:
# Calculate the standard deviation of the weekly cost
std_deviation = units_coefficient * std_deviation_per_unit

In [ ]:
# Calculate the test statistic (t)
t_statistic = (sample_mean - theoretical_mean) / (std_deviation / (sample_size ** 0.5))

In [ ]:
print(f'Test Statistic (t): {t_statistic}')

Test Statistic (t): -38.0


In [ ]:
# Determine the critical value for alpha level of 0.05 (one-tailed test)
alpha = 0.05
critical_value = stats.norm.ppf(1 - alpha)

In [ ]:
print(f'Critical Value (Z): {critical_value}')

Critical Value (Z): 1.6448536269514722


Since,


**t < Z
0.05
​
 , we reject the null hypothesis.**

 **There is strong evidence to support the restaurant owners' claim that the weekly operating costs are higher than the model suggests.**

# CHI-SQUARE TESTING

**Null Hypothesis (H₀):** There is no significant association between the type of smart home device purchased and the customer satisfaction level.

**Alternative Hypothesis (H₁):** There is a significant association between the type of smart home device purchased and the customer satisfaction level.

In [ ]:
observed = np.array([
    [50, 70],
    [80, 100],
    [60, 90],
    [30, 50],
    [20, 50]
])

print(observed)

[[ 50  70]
 [ 80 100]
 [ 60  90]
 [ 30  50]
 [ 20  50]]


In [ ]:
row_sums = observed.sum(axis=1)
col_sums = observed.sum(axis=0)
total = observed.sum()

# Expected frequencies
expected = np.outer(row_sums, col_sums) / total

In [ ]:
# Chi-Square statistic
chi_square_statistic = ((observed - expected) ** 2 / expected).sum()
print(f'Chi-Square Statistic: {chi_square_statistic}')

Chi-Square Statistic: 5.638227513227513


In [ ]:
# Degrees of freedom
df = (observed.shape[0] - 1) * (observed.shape[1] - 1)
print(f'Degrees of Freedom: {df}')

Degrees of Freedom: 4


In [ ]:
# Critical value
alpha = 0.05
critical_value = stats.chi2.ppf(1 - alpha, df)
print(f'Critical Value: {critical_value}')
# P-value
p_value = 1 - stats.chi2.cdf(chi_square_statistic, df)
print(f'P-value: {p_value}')

Critical Value: 9.487729036781154
P-value: 0.22784371130697179


In [ ]:
# Decision
if chi_square_statistic > critical_value:
    print("Reject the null hypothesis: There is a significant association between the type of smart home device purchased and the customer satisfaction level.")
else:
    print("Fail to reject the null hypothesis: There is no significant association between the type of smart home device purchased and the customer satisfaction level.")


Fail to reject the null hypothesis: There is no significant association between the type of smart home device purchased and the customer satisfaction level.


Since the computed Chi-Square statistic 5.63 is greater than the critical value of 9.488, and the p-value is less than 0.05, **we reject the null hypothesis.**

**There is strong evidence to suggest that there is a significant association between the type of smart home device purchased and the customer satisfaction level.**